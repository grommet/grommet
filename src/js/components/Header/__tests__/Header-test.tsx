import React from 'react';
import { render } from '@testing-library/react';

import { Grommet, Header } from '../..';

describe('Header', () => {
  it('should forward a ref to the underlying header', () => {
    const ref = React.createRef<HTMLDivElement>();

    render(
      <Grommet>
        <Header ref={ref}>Hello, World!</Header>
      </Grommet>,
    );

    expect(ref.current).not.toBe(null);
    expect(ref.current?.nodeName).toBe('HEADER');
  });
  it('should render outside grommet wrapper', () => {
    const { container } = render(<Header>Hello, World!</Header>);
    expect(container.firstChild).toMatchSnapshot();
  });
  test('theme gap', () => {
    const customTheme = {
      header: {
        gap: 'large',
      },
    };

    const { container } = render(
      <Grommet theme={customTheme}>
        <Header>
          <div>Header Item 1</div>
          <div>Header Item 2</div>
          <div>Header Item 3</div>
        </Header>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
