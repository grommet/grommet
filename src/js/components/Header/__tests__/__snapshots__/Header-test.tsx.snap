// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render outside grommet wrapper 1`] = `
<header
  class="StyledBox-sc-13pk1d4-0 iqBrqC"
>
  Hello, World!
</header>
`;
